### PR TITLE
Artifacts/rename

### LINKED ISSUE #448: rename artifacts to the names used in the release
- [x] (for mac) ship only `falcon-sql-client_2.7.0.dmg`
- [x] (for win) ship only `falcon-sql-client_2.7.0.exe`

### PR DESCRIPTION
**Pull request by n-riesco**
- documented steps to create a release (adds note to warn against using Firefox to download the artifacts from CircleCI; see #471)

- added version numbers to the installers filenames (closes #448)

---

@tarzzz would you review this PR?

### CODE REVIEW

File: circle.yml [hunk 1]
```diff
@@ -35,7 +35,6 @@ jobs:
       - run: yarn run test
       - run: yarn run pack
       - run: mkdir artifacts
-      - run: mv release/*.yml artifacts || true
-      - run: mv release/falcon-sql-client* artifacts
+      - run: for f in release/falcon-sql-client-*.tar.gz ; do mv $f artifacts/linux-falcon-v${f#release/falcon-sql-client-}; done
```
Comment (tarzzz) [inline circle.yml]: `{f#release/falcon-sql-client-}`?

File: NEW_RELEASE.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+- [ ] Build deb installer manually
+- [ ] Confirm deb installer works and Falcon is able to connect to sqlite
+- [ ] Build AppImage installer manually
+- [ ] Build rpm installer manually
+- [ ] Rebase and merge into `master`
+- [ ] Tag current `master` as `vM.m.p` and annotate it with the changelog message (this should automatically update the latest release in github)
+- [ ] Edit the release to attach all the installers.
+- [ ] Check links in https://plot.ly/free-sql-client-download/
+
+## Additional Checklist for an OnPrem Release
```
Comment (tarzzz) [inline NEW_RELEASE.md]: Probably remove this section from here? It is pretty-straightforward, and is not applicable for open-source Falcon.
Moreover Falcon-specific developers will not have access to the Quay builds anyways..!!

File: circle.yml (hunk 1 above)
Comment (n-riesco) [reply circle.yml]: it's a bash parameter substitution; e.g. see see https://www.tldp.org/LDP/abs/html/parameter-substitution.html

Comment (n-riesco) [reply circle.yml]: shall I perhaps add a comment explaining this line moves the installer from the `release` folder to `artifacts`?

File: NEW_RELEASE.md (hunk 2 above)
Comment (n-riesco) [reply NEW_RELEASE.md]: It's more convenient for me to have the full checklist.

File: circle.yml (hunk 1 above)
Comment (tarzzz) [reply circle.yml]: Yes, that would be great actually.. !!

File: NEW_RELEASE.md (hunk 2 above)
Comment (tarzzz) [reply NEW_RELEASE.md]: Ok, if it is convenient for you .. 😸

**Pull request merged**